### PR TITLE
doc: doc structure edits

### PR DESCRIPTION
**Pull request by greg-fer**
Updated landing page for Guides with structure from guides/README.md.
Excluded guides/README.md from doxygen build, so it only shows in the GitHub markdown preview.
Added links to docs on the main README.md, so that the doc page is possible to be found.

### CODE REVIEW

File: docs/guides/index.md [hunk 1]
```diff
@@ -6,7 +6,56 @@ and features.
...
+-   [Avahi - Troubleshooting](./troubleshooting_avahi.md)
+
+## Tool Guides
+
+-   [chip-tool](./chip_tool_guide.md)
+-   [Python Matter-Repl](./matter-repl.md)
+-   [python-chip-controller - Advanced](./python_chip_controller_advanced_usage.md)
+-   [python-chip-controller - Building](./python_chip_controller_building.md)
+
+## Build Guides
```
Comment (LuDuda) [inline docs/guides/index.md:59]: nit: Taking to account the importance.. can we move Build Guides before Platform Guides and Tool Guides just after Platform Guides?

**Review by LuDuda — APPROVED** [review]
LGTM :+1:

File: docs/guides/index.md (hunk 1 above)
Comment (greg-fer) [reply docs/guides/index.md:59]: Yes, although I'd like to do this in a separate PR, with some other planned changes.

File: README.md [hunk 2]
```diff
@@ -14,14 +16,26 @@
...
 [![Cirque](https://github.com/project-chip/connectedhomeip/workflows/Cirque/badge.svg)](https://github.com/project-chip/connectedhomeip/actions/workflows/cirque.yaml)
 [![QEMU](https://github.com/project-chip/connectedhomeip/workflows/QEMU/badge.svg)](https://github.com/project-chip/connectedhomeip/actions/workflows/qemu.yaml)
 
+**Tools**
+
 [![ZAP Templates](https://github.com/project-chip/connectedhomeip/workflows/ZAP/badge.svg)](https://github.com/project-chip/connectedhomeip/actions/workflows/zap_templates.yaml)
 
+**Documentation**
+
+-   [Documentation Build](https://github.com/project-chip/connectedhomeip/actions/workflows/docbuild.yaml)
```
Comment (Damian-Nordic) [inline README.md:35]: `[Documentation Build](https://github.com/project-chip/connectedhomeip/actions/workflows/docbuild.yaml)` to keep it consistent with other builds?

**Review by Damian-Nordic — APPROVED** [review]

File: docs/guides/index.md (hunk 1 above)
Comment (greg-fer) [reply docs/guides/index.md]: Actually, applied already here.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**